atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/generated/StreamTestWithHashStreamStore.java

```diff
@@ -214,13 +214,22 @@ public InputStream loadStream(Transaction t, final Long id) {
 
     @Override
     protected void tryWriteStreamToFile(Transaction transaction, Long id, StreamMetadata metadata, FileOutputStream fos) throws IOException {
-        try (InputStream blockStream = makeStream(id, metadata);
+        try (InputStream blockStream = eagerlyLoadStream(transaction, id, metadata);
                 InputStream decompressingStream = new LZ4BlockInputStream(blockStream);
                 OutputStream fileStream = fos;) {
             ByteStreams.copy(decompressingStream, fileStream);
         }
     }
 
+    private InputStream eagerlyLoadStream(Transaction parent, Long id, StreamMetadata metadata) {
```

Comment:
The nice thing about the previous implementation (using makeStream) is that we actually stream from cassandra to the file. This implementation holds the entire stream contents (albeit compressed) in memory.